### ISSUE TITLE
Feature request: Export disassembly to text

### ISSUE THREAD

**Issue by comex** [Vector35/binaryninja-api#2502]
**Is your feature request related to a problem? Please describe.**
I want to do some simple custom pattern matching on a program's assembly instructions.

**Describe the solution you'd like**
I would like to be able to export a disassembly of the entire program, including addresses, labels, etc., to a text file so I can then use standard text tools to analyze or share it.

**Describe alternatives you've considered**
- Select All + Copy.  However, Select All was greyed out.  I tried selecting the entire program by clicking at the beginning then shift-clicking at the end, then pressing copy; however, this only copied about a screen or two's worth of diassembly, not the entire program.
- I could do the pattern matching using BN's Python API.
- I could implement export-to-text using BN's Python API.
- I could use a different disassembler on the same file.
In practice I will probably do the last one since I'm lazy.

**Additional context**
Bells and whistles that would be nice to have along with this:
- Option to export any of the IL views in addition to disassembly
- Option to export with syntax highlighting in some suitable format (HTML or RTF)

IDA has two ways to export a disassembly to text.  The one I mainly used, 'Create LST File', exports the same text seen in the interface.  There is also 'Create ASM File' which I've never had a use for, but apparently excludes the address column so that it's closer to something that could be passed back to an assembler.  Beyond that, there is 'Create C File' to export a decompilation.

**Comment by comex** [comment]
(in before someone points out there's some obvious way to do this already…)

**Comment by KyleMiles** [comment]
FWIW, @psifertex has a snippet that does some text export here: https://gist.github.com/psifertex/6fbc7532f536775194edd26290892ef7#file-save_decompilation-py

**Comment by psifertex** [comment]
Yeah, sorry for not chiming in sooner. I was actually working on a more fully featured bit of code during a vacation for a side-task, but forgot to finish it off. Part of the hold up is we're working on "display as C" in #1650 and I kinda wanted to wait and see if that changed any of the display APIs for output style before I finished off the plugin.

Anyway, I just finished the changes and it's available in: https://gist.github.com/psifertex/6fbc7532f536775194edd26290892ef7#file-export_to_text-py
